Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -259,74 +259,93 @@ namespace AzToolsFramework
 
                         PrefabDom reparentPatch;
-                        m_instanceToTemplateInterface->GeneratePatch(reparentPatch, nestedContainerEntityDomBefore, nestedContainerEntityDomAfter);
-                        m_instanceToTemplateInterface->AppendEntityAliasToPatchPaths(reparentPatch, nestedInstanceContainerEntityId);
+                        m_instanceToTemplateInterface->GeneratePatch(
+                            reparentPatch, nestedContainerEntityDomBefore, nestedContainerEntityDomAfter);
+                        m_instanceToTemplateInterface->AppendEntityAliasToPatchPaths(reparentPatch, topLevelEntityId);
 
                         // We won't parent this undo node to the undo batch so that the newly created template and link will remain
                         // unaffected by undo actions. This is needed so that any future instantiations of the template will work.
-                        PrefabUndoUpdateLink linkUpdate = PrefabUndoUpdateLink(AZStd::to_string(static_cast<AZ::u64>(nestedInstanceContainerEntityId)));
-                        linkUpdate.Capture(reparentPatch, nestedInstance->GetLinkId());
+                        PrefabUndoUpdateLink linkUpdate = PrefabUndoUpdateLink(AZStd::to_string(static_cast<AZ::u64>(topLevelEntityId)));
+                        linkUpdate.Capture(reparentPatch, topLevelInstance->get().GetLinkId());
                         linkUpdate.Redo();
                     }
+                    else
+                    {
+                        AZ::TransformBus::Event(topLevelEntityId, &AZ::TransformBus::Events::SetParent, newContainerEntityId);
+
+                        // Update the non-container entity DOM in the new template with the new parent transform data.
+                        // This is done without undo/redo support so that any future instantiations of the template will work.
+                        PrefabDom topLevelEntityDomWithParentInTransform(&(newInstanceTemplateDom.GetAllocator()));
+                        m_instanceToTemplateInterface->GenerateEntityDomBySerializing(
+                            topLevelEntityDomWithParentInTransform, *topLevelEntity);
+
+                        PrefabDomPath entityAliasDomPath(m_instanceToTemplateInterface->GenerateEntityAliasPath(topLevelEntityId).c_str());
+                        entityAliasDomPath.Set(newInstanceTemplateDom, topLevelEntityDomWithParentInTransform.Move());
+                    }
+                }
+                EditorEntityContextNotificationBus::Broadcast(&EditorEntityContextNotification::SetForceAddEntitiesToBackFlag, false);
+
+                // Set up remaining links for the nested instances that are not at the top level. This is done without undo/redo support.
+                instanceToCreate->get().GetNestedInstances([topLevelEntities, createLinkForNestedInstanceFunc](AZStd::unique_ptr<Instance>& nestedInstance) {
```

Comment:
I assume this piece of code handles case like below when a prefab is not a toplevel entity but a child of another entity? If that's the case, it would be good to add another unit test to cover this logic as it won't be covered with the test you added below

- entity1 (select)
  - prefab1(select)
- entity2 (select)

Reply:
Actually thinking about it more, the entity order only applies to top-level entities, so prefab1 wouldn't be reparented in this case.

Reply:
Yes, it shouldn't be reparented, it should retain its original parent. It'll be good to validate that still happens though. Because the code highlighted is responsible for creating a link between 'prefab1' and the new prefab that is being created now. So, a new link will be created for 'prefab1' but it should retain its old entity parent. It will be a good test case to have because of the way the code is split in the CreatePrefab() function

Reply:
+1. I also think we can, probably in a follow-up PR, add one more test to validate parent entities of non-top-level entities. 

For 'prefab1', we can also add these to validate the change in its owning instance.
```cpp
ValidateNestedInstanceNotUnderInstance(<previous parent instance>, prefab1InstanceAlias);
ValidateNestedInstanceUnderInstance(<newly created prefab instance>, prefab1InstanceAlias);
```

Reply:
I agree we need to add more "create prefab" tests. I will create a separate ticket for that.